projects/RabbitMQ.Client/client/api/ConnectionConfig.cs

```diff
@@ -127,7 +137,7 @@ public sealed class ConnectionConfig
             UserName = userName;
             Password = password;
             AuthMechanisms = authMechanisms;
-            ClientProperties = clientProperties;
+            ClientProperties = new ReadOnlyDictionary<string, object?>(clientProperties);
```

Comment:
Do we really want a defensive copy here? Also why for the properties but not for the authMechanisms? I'd not do it at all.

Reply:
Please see this code:

https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1110/files#diff-09ed630152a341b50003a5c1da0fb23eecb62eec4df30df695f084bd14f52760R82-R86

Reply:
I'm not sure what you're trying to show me with that code.

Question is, do we want to protect all incoming collections from modifications from outside, then we need to do a defensive copy for all of them. Or we trust that due to the ReadOnly Collection Interfaces they will not be modified.

I personally would argue that if you modify a provided read only collection, you're doing something you're not supposed to do, so therefore I would penalize all user that do it correctly by a defensive copy.

Reply:
I don't believe that constructor copies anything ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlydictionary-2.-ctor?view=net-6.0#system-collections-objectmodel-readonlydictionary-2-ctor(system-collections-generic-idictionary((-0-1)))).

At this point you've convinced me it's overkill at this point to make `ConnectionConfig` immutable. I'll revert my changes.

Reply:
True, didn't know that. Thanks for linking it.